File: HW11/Q5.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main()
{
    int n, q;
    cin >> n >> q;

    vector<int> a(n);
    for (int i = 0; i < n; i++)
        cin >> a[i];

    for (int i = 0; i < q; i++)
    {
        int op, l, r;
        cin >> op >> l >> r;

        if (op == 1)
        {
            int result = 0;
            int prev = 0;
            l = l - 1;

            for (int j = l; j < r; j++)
            {
                prev = min(prev + a[j], a[j]);
                result = min(result, prev);
            }

            cout << result << endl;
        }
        else if (op == 2)
        {
            a[l - 1] = r;
        }
    }

    return 0;
}

File: HW11/Q5.py
n, q = map(int, input().split())
a = list(map(int, input().split()))

for _ in range(q):
    op, l, r = map(int, input().split())
    if op == 1:
        result = 0
        prev = 0
        l = l-1
        for i in range(l, r):
            result = min(result, prev+a[i])
            prev = min(prev+a[i], 0)
        print(result)
    elif op == 2:
        a[l - 1] = r
